src/getexchange.c

```diff
@@ -262,6 +262,19 @@ int getexchange(void) {
 		break;
 	    }
 
+	    case KEY_UP:	/* Up/Down--increase/decrease serial number */
+	    case KEY_DOWN: {
+		if (i > 0 && i <= 5 && strspn(comment, "0123456789") == i) { /* comment non-empty and only digits */
+		    int nr = atoi(comment);
+		    nr += (x == KEY_UP) ? 1 : -1;
+		    if (nr >= 0 && nr <= 99999) {
+			sprintf(comment, "%0*d", i, nr); /* preserve leading zeros */
+			i = strlen(comment); /* length can change when overflowing 9 -> 10 */
```

Comment:
`i` must be `<= contest->exchange_width` even this is currently implicitly true. Exchange width may become configurable in the future and one may set it to <5.

Reply:
Updated.